### PR TITLE
Include platform in SharedFx bundle name

### LINKED ISSUE #32708: Missing x86 vs x64 qualifier in add/remove programs for ASP.NET Core shared framework
### Describe the bug
ASP.NET Core Shared Framework add/remove programs entry doesn't differentiate x86 and x64 and appears like a duplicate.

image

### To Reproduce

I saw this after Microsoft Update pushed new versions to my machine. The prior versions were from Visual Studio and hidden from add/remove programs. I have not verified, but I suspect you can repro this on any machine by installing both x64 and x86 ASP.NET Core shared frameworks

### Further technical details
```
.NET SDK (reflecting any global.json):
 Version:   5.0.300-preview.21258.4
 Commit:    e659912dfb

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.300-preview.21258.4\

Host (useful for support):
  Version: 5.0.6
  Commit:  478b2f8c0e

.NET SDKs installed:
  5.0.300-preview.21258.4 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

### PR DESCRIPTION
**Pull request by wtgodbe**
Fixes https://github.com/dotnet/aspnetcore/issues/32708

I'll use this PR as a guinea pig for the backport bot introduced by https://github.com/dotnet/aspnetcore/pull/32777

Internal build to grab installers for validation: https://dev.azure.com/dnceng/internal/_build/results?buildId=1142108&view=results

### CODE REVIEW

**Review by joeloff — APPROVED** [review]

**Comment by wtgodbe** [comment]
sfxx64
Looks like that worked

File: src/Installers/Windows/SharedFrameworkBundle/SharedFrameworkBundle.wixproj [hunk 1]
```diff
@@ -66,8 +66,8 @@
 
     <BundleNameShort>Microsoft ASP.NET Core $(PackageBrandingVersion)</BundleNameShort>
     <BundleNameSub>Shared Framework</BundleNameSub>
-    <BundleName>$(BundleNameShort) - $(BundleNameSub)</BundleName>
-    <BundleNameFull>$(BundleName) ($(Platform))</BundleNameFull>
+    <BundleName>$(BundleNameShort) - $(BundleNameSub) ($(Platform))</BundleName>
+    <BundleNameFull>$(BundleName)</BundleNameFull>
     <BundleRegName>$(BundleNameFull)</BundleRegName>
```
Comment (dougbu) [inline src/Installers/Windows/SharedFrameworkBundle/SharedFrameworkBundle.wixproj:71]: Why keep all of `$(BundleName)`, `$(BundleNameFull)` and `$(BundleRegName)` if their values should be the same❔

More importantly, don't we have the same problem w/ the Windows Hosting bundle https://github.com/dotnet/aspnetcore/blob/52eff90fbcfca39b7eb58baad597df6a99a542b0/src/Installers/Windows/WindowsHostingBundle/WindowsHostingBundle.wixproj#L84-L85❔

**Review by dougbu — APPROVED** [review]

File: src/Installers/Windows/SharedFrameworkBundle/SharedFrameworkBundle.wixproj (hunk 1 above)
Comment (wtgodbe) [reply src/Installers/Windows/SharedFrameworkBundle/SharedFrameworkBundle.wixproj:71]: I guess they correspond to specific wix properties - we could just have one msbuild property that the .wxs file references three times, but I don't think it matters to much. As for the hosting bundle, that's platform-agnostic, so it doesn't need to specify what platform it's for in the name.

**Pull request merged**